data/adapters/reactome_pathway_adapter.py

```diff
@@ -0,0 +1,111 @@
+from adapters import Adapter
+import gzip
+import json
+import os
+from db.arango_db import ArangoDB
+import requests
+from requests.adapters import HTTPAdapter, Retry
+from requests.exceptions import JSONDecodeError
+
+
+# Example pathway input file:
+# R-GGA-199992	trans-Golgi Network Vesicle Budding	Gallus gallus
+# R-HSA-164843	2-LTR circle formation	Homo sapiens
+# R-HSA-73843	5-Phosphoribose 1-diphosphate biosynthesis	Homo sapiens
+# R-HSA-1971475	A tetrasaccharide linker sequence is required for GAG synthesis	Homo sapiens
+# R-HSA-5619084	ABC transporter disorders	Homo sapiens
+
+
+class ReactomePathway(Adapter):
+
+    OUTPUT_FOLDER = './parsed-data'
+
+    def __init__(self, filepath=None, dry_run=False):
+
+        self.filepath = filepath
+        self.label = 'pathway'
+        self.dataset = 'pathway'
+        if not os.path.exists(ReactomePathway.OUTPUT_FOLDER):
+            os.makedirs(ReactomePathway.OUTPUT_FOLDER)
+        self.output_filepath = '{}/{}.json'.format(
+            ReactomePathway.OUTPUT_FOLDER,
+            self.dataset,
+        )
+        self.dry_run = dry_run
+        self.SKIP_BIOCYPHER = True
+
+        super(ReactomePathway, self).__init__()
+
+    def process_file(self):
+        parsed_data_file = open(self.output_filepath, 'w')
+        session = requests.Session()
+        retries = Retry(total=5, backoff_factor=1,
+                        status_forcelist=[500, 502, 503, 504])
+        session.mount('https://', HTTPAdapter(max_retries=retries))
+
+        with open(self.filepath) as input:
+            for line in input:
+                id, name, species = line.strip().split('\t')
+                if species == 'Homo sapiens':
+                    to_json = {
+                        '_key': id,
+                        'name': name,
+                        'species': species,
+                        'source': 'Reactome',
+                        'source_url': 'https://reactome.org/'
+                    }
+
+                    try:
+                        query = 'https://reactome.org/ContentService/data/query/' + id
```

Comment:
I'm thinking of ways of avoiding this, so we don't depend on their API to be functional... a few options I thought of:

* do they offer any data download that we can keep somewhere?
* run this script once and save all responses in a file and keep that versioned

do you have any other ideas?